### ISSUE TITLE
Host related problems

### ISSUE THREAD

**Issue by repetier** [prusa3d/Prusa-Firmware-Buddy#369]
While testing how well the printer plays with our repetier-server software I encountered some issues I want to mention here. All are tested with latest firmware 4.0.5.

When I tried to change filament over server I use M702 which works great, but M701 for loading fails. It takes the element and purges. And after that I only get busy messages but nothing happens. That is the point where when doing it via printer display I would get the question if color did change, so guess that is what firmware is waiting for, but the dialog is not open so the only way to proceed from there is to reset and print is ruined.

The next issue is the filament sensor. If we do NOT send M876 P1 the firmware will ignore the out of filament signal. Moreover if you would say it is up to server to handle it - ok with that, but there is NO message to tell host software that we ran out of filament. The standard message for this would be
// action: out_of_filament T0
If you handle it internally, you would just send a busy signal and offer change. Both is ok but no message and no action is worst that can happen.

It is only handled internally if I send
M876 P1
which is not really a requirement I think. Especially since this shows that host can handle dialogs internally as described here:
https://reprap.org/wiki/G-code#M876:_Dialog_handling
But you do not send the dialog and handle it internally when host software says it can handle dialogs. And we currently do not support dialogs so not sending that. That the dialog for change not succeeds is an already known issue I think.

More over if I would send M876 P1 I would get a very discriminating view on display showing the octoprint logo. While Octoprint is a good software it is not the only one that can talk with your printer. I would really like a more neutral image here.

I think you should not make that view dependent on M876 at all apart from being neutral. It prevents the menu all the time while blocking them makes only sense during a print. And if you show it showing some progress info would be nice for your users. You could still get them via M73. If you implement PROGRESS capability https://reprap.org/wiki/Firmware_Capabilities_Protocol#Cap:PROGRESS
you would be able to show/hide the view by scanning the M530 command. I think there is no reason to not accept even both formats - M73 and the PROGRESS command group.

Apart from this it works well, also I would have of course more wishes to make colaborative work with hosts work better. E.g. having line number with ok to detect lost "ok" faster and "wait" message when idle. These make ugly artifacts on communication errors more unlikely and allow a faster send of new commands. I know marlin supports these in advanced settings, but has them deactivated by default. Never understood that decision unless you want to prevent using hosts.

**Comment by danopernis** [comment]
@repetier Is this still relevant? The firmware got two major releases and there have been some progress w.r.t. serial printing. If the problem persists, it would be helpful to split this into several issues, each with a narrow scope. Also, there might already be some existing open issues so please avoid duplicating them if that's the case.
